sympy/core/expr.py

```diff
@@ -329,98 +329,66 @@ def __complex__(self):
         re, im = result.as_real_imag()
         return complex(float(re), float(im))
 
-    def __ge__(self, other):
-        from sympy import GreaterThan
+    def _cmp(self, other, op, cls):
+        assert op in ("<", ">", "<=", ">=")
         try:
             other = _sympify(other)
         except SympifyError:
-            raise TypeError("Invalid comparison %s >= %s" % (self, other))
+            raise TypeError("Invalid comparison %s %s %s" % (self, op, other))
         for me in (self, other):
             if me.is_extended_real is False:
                 raise TypeError("Invalid comparison of non-real %s" % me)
             if me is S.NaN:
                 raise TypeError("Invalid NaN comparison")
+
         n2 = _n2(self, other)
         if n2 is not None:
-            return _sympify(n2 >= 0)
+            if op == "<":
+                return _sympify(n2 < 0)
+            elif op == ">":
+                return _sympify(n2 > 0)
+            elif op == "<=":
+                return _sympify(n2 <= 0)
+            else: # >=
+                return _sympify(n2 >= 0)
+
         if self.is_extended_real and other.is_extended_real:
-            if (self.is_infinite and self.is_extended_positive) \
-                    or (other.is_infinite and other.is_extended_negative):
-                return S.true
-            nneg = (self - other).is_extended_nonnegative
-            if nneg is not None:
-                return sympify(nneg)
-        return GreaterThan(self, other, evaluate=False)
+            # this should work for infinite values as well,
+            # so we don't need a separate check
+            diff = self - other
+            if diff is not S.NaN:
+                if op == "<":
+                    test = sympify(diff.is_extended_negative)
```

Comment:
We should probably use `as_Boolean` rather than `sympify` here.

Reply:
But it could be None

Reply:
Actually sympify is called below if test is not None so we don't need to call sympify here and can use `as_Boolean` below.

Reply:
`as_Boolean` causes import troubles, but I should just be able to do `S.true if test == True else S.false`